src/Microsoft.AspNetCore.SignalR.Core/Internal/DefaultHubProtocolResolver.cs

```diff
@@ -24,7 +24,7 @@ public IHubProtocol GetProtocol(string protocolName, HubConnectionContext connec
                 case "json":
                     return new JsonHubProtocol(JsonSerializer.Create(_options.Value.JsonSerializerSettings));
                 case "messagepack":
-                    return new MessagePackHubProtocol();
+                    return new MessagePackHubProtocol(_options.Value.MessagePackSerializationContext);
```

Comment:
If user doesn't add `HubOptions` then this can be null and `MessagePackHubProtocol` wont have any serialization settings.

Reply:
Seems to work just fine. [OptionsFactory](https://github.com/aspnet/Options/blob/97a81e4603e42d6d8ad3539ba3255cce2bc57fc4/src/Microsoft.Extensions.Options/OptionsFactory.cs#L30) creates options and we initialize the `MessagePackSerializationContext` property to a default value when HubOptions instance is created.